client/protocol.go

```diff
@@ -256,34 +259,29 @@ func (c *protocolClient) fastRound(cmd *xlineapi.Command) (*fastRoundRes, error)
 	for i := 0; i < len(c.connects); i++ {
 		select {
 		case res := <-resCh:
-			c.state.checkAndUpdate(*res.LeaderId, res.Term)
-			switch pr := res.ExeResult.(type) {
-			case *curpapi.ProposeResponse_Result:
-				okCnt++
-				switch cr := pr.Result.Result.(type) {
-				case *curpapi.CmdResult_Ok:
-					if isLeaderOK {
-						panic("should not set exe result twice")
-					}
-					isLeaderOK = true
-					err := proto.Unmarshal(cr.Ok, &exeResult)
-					if err != nil {
-						panic(err)
-					}
-				case *curpapi.CmdResult_Error:
-					err := proto.Unmarshal(cr.Error, &exeErr)
-					if err != nil {
-						panic(err)
-					}
-					c.logger.Info("fast round failed", zap.Any("propose ID", cmd.ProposeId))
-					return nil, &CommandError{err: &exeErr}
+			okCnt++
+			if res.GetResult() == nil {
+				continue
+			}
+			switch cmdRes := res.GetResult().Result.(type) {
+			case *curpapi.CmdResult_Ok:
+				if isLeaderOK {
+					panic("should not set exe result twice")
 				}
-			case *curpapi.ProposeResponse_Error:
-				c.logger.Warn("propose fail", zap.Error(errors.New(pr.Error.String())), zap.Any("propose ID", cmd.ProposeId))
-			default:
-				okCnt++
+				isLeaderOK = true
+				err := proto.Unmarshal(cmdRes.Ok, &exeResult)
+				if err != nil {
+					panic(err)
+				}
+			case *curpapi.CmdResult_Error:
+				err := proto.Unmarshal(cmdRes.Error, &exeErr)
+				if err != nil {
+					panic(err)
+				}
+				return nil, &CommandError{err: &exeErr}
 			}
 		case err := <-errCh:
+			// TODO: error handling
```

Comment:
You're right, error handling is necessary. It will be implemented in the next few pr's before the next release. The reason for not adding it to this pr is to ensure that each pr does only one thing.

Reply:
Error handling should also be part of this refactoring, and since the code size of these two PRs is small, I don't think we need to split the two PRs

Reply:
I can't agree more. Error handling should not be a new feat. We should add it in this pr.

Reply:
You're right, I've added it to this pr.